src/native/profiler.cc

```diff
@@ -267,7 +270,6 @@ void Profiler::runExperiment(JNIEnv * jni_env) {
   logger->flush();
 
   delete[] current_experiment.location_ranges;
-  free(sig);
```

Comment:
I think this should actually be `jvmti->Deallocate(sig)` as it's allocated with `jvmti->GetClassSignature` which according to https://docs.oracle.com/javase/8/docs/platform/jvmti/jvmti.html#GetClassSignature needs to be freed with `Deallocate`.

Reply:
Oh, I see, you free it in `bci_hits::create_dump`. Let's remove that `free` and instead `free` it here with `jvmti->Deallocate`. I'd rather keep memory management for a buffer in the same scope when possible.